aitextgen/train.py

```diff
@@ -172,10 +176,19 @@ def on_batch_end(self, trainer, pl_module):
                 desc += f" — GPU Mem: {gpu_memory} MB"
             self.main_progress_bar.update(self.progress_bar_refresh_rate)
             self.main_progress_bar.set_description(desc)
-
+        
+        if _TPU_AVAILABLE and self.save_every_check:
```

Comment:
Could we get some more information around this and why it's necessary? I think we can iterate on this to get something cleaner; the idea for lightnign is that this boilerplate can live within lightning outside of the user code.